src/idom/utils.py

```diff
@@ -221,5 +272,36 @@ def _hypen_to_camel_case(string: str) -> str:
     return first.lower() + remainder.title().replace("-", "")
 
 
-class HTMLParseError(etree.LxmlSyntaxError):  # type: ignore[misc]
-    """Raised when an HTML document cannot be parsed using strict parsing."""
+# Pattern for delimitting camelCase names (e.g. camelCase to camel-case)
+_CAMEL_CASE_SUB_PATTERN = re.compile(r"(?<!^)(?=[A-Z])")
+
+# see list of HTML attributes with dashes in them:
+# https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#attribute_list
+_CAMEL_TO_DASH_CASE_HTML_ATTRS = {
+    "acceptCharset": "accept-charset",
+    "httpEquiv": "http-equiv",
+}
+
+
+def _vdom_to_html_attr(key: str, value: Any) -> tuple[str, str]:
+    if key == "style":
+        if isinstance(value, dict):
+            value = ";".join(
+                # We lower only to normalize - CSS is case-insensitive:
+                # https://www.w3.org/TR/css-fonts-3/#font-family-casing
+                f"{_CAMEL_CASE_SUB_PATTERN.sub('-', k).lower()}:{v}"
+                for k, v in value.items()
+            )
+    elif key.startswith("data"):
+        # Handle data-* attribute names
+        key = _CAMEL_CASE_SUB_PATTERN.sub("-", key)
+    else:
+        key = _CAMEL_TO_DASH_CASE_HTML_ATTRS.get(key, key)
```

Comment:
Would be more future proof to convert all attributes be converted from `camelCase` to `kebab-case`, rather than just `acceptCharset` and `httpEquiv`.

Reply:
The trouble is that for cases like `tabIndex`, the conversion is `tabindex`, not `tab-index`. Not really sure there's a great alternative.

Reply:
I can re-writing it, but the current implementation only converts if the key is present in the mapping.